lib/index.ts

```diff
@@ -463,6 +482,41 @@ export class DockerProgress {
 		return hash;
 	}
 
+	/** Build docker image calling onProgress with extended progress info regularly */
+	async build(
+		tarStream: Stream.Readable,
+		onProgress: ProgressCallback,
+		options?: BuildOptions,
+	): Promise<string> {
+		const ignoreErrorEvents = !!options?.ignoreProgressErrorEvents;
+
+		// Authconfig is not supported by docker `/build` endpoint but we support
+		// it here to provide a common interface with the `pull()` method
+		if (options?.authconfig) {
+			const { serveraddress, ...authconfig } = options.authconfig;
+			options.registryconfig = {
+				[serveraddress]: authconfig,
+
+				// Override with registryconfig if it the same serveraddress
+				// is used in both options
+				...(options.registryconfig ?? {}),
```

Comment:
NIT: The `?? {}` operation is not needed for objects, unlike arrays:
```
$ node
> let o = {}
> o = { ...(undefined) }
{}
> let a = []
> a = [ ...(undefined) ]
Uncaught:
TypeError: undefined is not iterable (cannot read property Symbol(Symbol.iterator))
```
```suggestion
				...options.registryconfig,
```

Reply:
nice, even cleaner then